### DISCUSSION TITLE
Servizi inerenti i provvedimenti di rifiuto di una notifica

### DISCUSSION THREAD

**Discussion by MichelaGarzoni** [italia/ansc#1045] [Supporto servizi cooperativi]
Buongiorno, 
ho alcuni dubbi in relazione ai provvedimenti di rifiuto delle notifiche in quanto non riesco ad evidenziare quali servizi cooperativi utilizzare per poter eseguire le stesse cose che vengono presentate lato web-app e non trovo documentazione in merito a questo processo. 

In elenco i miei dubbi:

1- sulla web-app, dopo il primo rifiuto di una notifica, mi viene chiesto di inserire un provvedimento: a questo punto, utilizzando il servizio cooperativo [R008]/notifiche/ricerca il campo idStatoFlussoNotifiche assume il valore 3(rifiutato). Se inserisco il provvedimento di rifiuto e rieseguo la ricerca della notifica, questa presenta sempre il valore 3 nel campo idStatoFlussoNotifiche. Come faccio a distinguere lato servizio cooperativo se per la notifica è stato inserito già il provvedimento di rifiuto o meno?

2- [R008]/notifiche/ricerca: impostando un intervallo temporale come filtro di ricerca, ottengo un elenco di notifiche. Per le notifiche relative alle "annotazioni automatiche" ho presente l'idAnsc dell'annotazione, ma non ha nessun campo che faccia riferimento all'evento per cui è stata generata tale annotazione. Come faccio a collegare tale annotazione all'evento?

3- Web-App inserimento provvedimento di rifiuto: oltre ai campi "motivazione rifiuto" e "numero protocollo provvedimento rifiuto" richiede di inserire l'intestatario (un eventuale secondo intestatario) e gli allegati relativi al provvedimento di rifiuto, i quali sono da verificare e convalidare. A termine si passa alla validazione del provvedimenti di rifiuto. Sui servizi cooperativi, il servizio di validazione del provvedimento di rifiuto da utilizzare è l'[R016]/rifiuto/validazione/evento? in tal caso, come mai nello yaml non trovo la struttura ed i campi da popolare relativi agli allegati inviati e agli intestatari? E' errata la mia interpretazione su quale servizio utilizzare? Qual è la sequenza corretta dei servizi da richiamare?
Grazie

**Comment by asqueglia** [comment]
Buongiorno, provo a rispondere di seguito alle sue domande.

1. non è obbligatorio inserire un provvedimento di rifiuto a fronte del rifiuto di una notifica; non è stato previsto quindi il cambio di stato della notifica a seguito dell'inserimento del provvedimento di rifiuto.
2. con il rilascio della versione 1.38.0  del 04-03-2025, sono state estese le notifiche relative alle annotazioni automatiche inserendo oltre al comune di redazione dell'atto che ha generato la annotazione anche il comune di destinazione dell'atto su cui va apposta l'annotazione; le informazioni dell'atto di destinazione e relativo comune vengono restituite nel  il campo 'attoPrimario'.
3. si conferma che il servizio per l'inserimento del provvedimento di rifiuto è il '[R016]/rifiuto/validazione/evento; la request del servizio richiede di trasmettere l'evento come dataModelEvento; i dati obbligatori da trasmettere sono l'intestatario e i datiRifiuto; a fine mese verrà rilasciato u  controllo bloccante che obbligherà appunto a trasmettere obbligatoriamente l'intestatario del  rifiuto. Di seguito un payload di esempio 

`"evento": {
        "idTipoEvento": "1",
        "idtipocontenuto": "5",
        "descrizioneCasoUso": "Provvedimento di Rifiuto",
        "stato": "CONFERMATO",
        "dataformazione": "2025-03-03",
        "ora": "09",
        "minuto": "25",
        "annoAtto": "2025",
        "idComuneRegistrazione": "1000000",
        "nomeComuneRegistrazione": "ANPR",
        "idProvinciaRegistrazione": "58",
        "siglaProvinciaRegistrazione": "RM",
        "operatore": "CRUMRA67S47Z127B",
        "ruoloOperatore": "2",
        "operatoreNome": "MARIE",
        "operatoreCognome": "CURIE",
        "redattoNellaCasaComunale": true,
        "allegati": [
            {
                "id": 597584,
                "idTipoAllegato": 998,
                "hashAllegato": "kuuyGfA/k6bmtOSe31s3PtsA87HBTKyKIW8Ll8nCg3k=",
                "descrizione": "a"
            }
        ],
        "intestatari": [
            {
                "id": "",
                "idANPR": "",
                "idsoggettoanpr": "",
                "codiceFiscale": "",
                "cognome": "padre",
                "nome": "padre",
                "sesso": "M",
                "dataNascita": "2000-03-03",
                "idStatoNascita": "2",
                "nomeStatoNascita": "ALBANIA",
                "idComuneResidenza": "",
                "nomeComuneResidenza": "",
                "idProvinciaResidenza": "",
                "siglaProvinciaResidenza": "",
                "idStatoResidenza": "",
                "nomeStatoResidenza": "",
                "indirizzoResidenza": "",
                "idNazionalita": "",
                "nazionalita": "",
                "flagDichiarante": false,
                "flagComparente": false,
                "documentoRiconoscimento": {},
                "localitaEsteraNascita": "",
                "idComuneAIRE": "",
                "nomeComuneAIRE": "",
                "idProvinciaAIRE": "",
                "siglaProvinciaAIRE": "",
                "idSoggetto": ""
            },
            {
                "flagDichiarante": false,
                "flagComparente": false,
                "documentoRiconoscimento": {}
            }
        ],
        "datains": "2025-03-03T08:23:50.494511",
        "dataultimoagg": "2025-03-03T08:25:27.402051",
        "flagsecretato": 0,
        "idVersion": 100021,
        "datiRifiuto": {
            "eventoCollegato": {
                "id": 584401,
                "idAnsc": "2025-584401-123-999999",
                "idComuneRegistrazione": "1000000",
                "nomeComuneRegistrazione": "ANPR",
                "idProvinciaRegistrazione": "58",
                "siglaProvinciaRegistrazione": "RM",
                "idtipocontenuto": "3",
                "dataformazione": "2025-03-03",
                "numeroatto": "123",
                "annoAtto": "2025",
                "tipologia": "1"
            },
            "idTipoRifiuto": 2,
            "motivazioneRifiuto": "aaa",
            "numeroProtocolloProvvedimento": "123123"
        }
    }
}`

Chiudiamo pertanto la discussion. In caso di ulteriori dubbi in merito si prega di aprirne una altra dedicata
